src/controller/CHIPDeviceController.h

```diff
@@ -229,7 +229,8 @@ class DLL_EXPORT DeviceController : public SessionRecoveryDelegate,
     CHIP_ERROR UpdateDevice(NodeId deviceId)
     {
         VerifyOrReturnError(mState == State::Initialized && mFabricInfo != nullptr, CHIP_ERROR_INCORRECT_STATE);
-        return mCASESessionManager->ResolveDeviceAddress(mFabricInfo, deviceId);
+        bool establishSession = true;
+        return mCASESessionManager->ResolveDeviceAddress(mFabricInfo, deviceId, establishSession);
```

Comment:
So this ends up notifying via `DeviceCommissioner::OnOperationalNodeResolved` if we happen to be a device commissioner, which then establishes a CASE session?  But if we're a device controller that is not a device commissioner ... then we don't?  It's a little odd to me that UpdateDevice will do different things depending on whether the controller is a commissioner or not.

Looking at the consumers, all of them except `OnFirstMessageDeliveryFailed` are working with a commissioner anyway, and that one arguably _would_ want to try establishing CASE, right?

So it seems like it might be clearer to just have UpdateDevice call `FindOrEstablishSession` and not have the strange indirection where we have it do a resolve, which then via the other delegate interface calls back into us, and then we kick off CASE.

I'm fine with this happening as a followup, but I would really like us to clean this up so it's clear what is going on instead of having the various action at a distance...